htdocs/api/v0.0.3-dev/candidates/Visit.php

```diff
@@ -179,13 +186,42 @@ public function handlePUT()
             $this->error("You are not affiliated with any site");
             $this->safeExit(0);
         } else if ($num_sites > 1) {
-            $this->header("HTTP/1.1 501 Not Implemented");
-            $this->error(
-                "This API version does not support timepoint creation " .
-                "by users with multiple site affiliations. This will be ".
-                "implemented in a future release."
-            );
-            $this->safeExit(0);
+            if (!isset($this->ReceivedJSON['Meta']['Site'])
+            ) {
+                $this->header("HTTP/1.1 400 Bad Request");
+                $this->error(
+                    "Users affiliated with multiple sites " .
+                    "need to specify the name of the site at " .
+                    "which the visit took place."
+                );
+                $this->safeExit(0);
+            } else {
+                $siteName = $this->ReceivedJSON['Meta']['Site'];
+                foreach ($centerIDs as $key => $centerID) {
```

Comment:
this loop could be cleaned up into a single query if you use @ridz1208 's `pselectColWithIndexKey` (this will require a rebase however)